File: backend/tests/fixtures/security/python/safe_app.py
import sqlite3
from flask import Flask, request

app = Flask(__name__)


@app.route("/search")
def search():
    q = request.args.get("q")
    conn = sqlite3.connect(":memory:")
    conn.execute("SELECT * FROM users WHERE name = ?", (q,))
    return "ok"


@app.route("/page")
def page():
    return "hello"

File: backend/tests/fixtures/security/javascript/safe_app.js
const express = require("express");
const { execFile } = require("child_process");
const app = express();

app.get("/run", (req, res) => {
  execFile("echo", ["hello"], (err, stdout) => {
    res.send(stdout);
  });
});

app.get("/search", (req, res) => {
  const q = req.query.q;
  db.query("SELECT * FROM users WHERE name = $1", [q]);
  res.send("ok");
});
